frontend/src/components/NewRunParameters.tsx

```diff
@@ -39,23 +61,145 @@ class NewRunParameters extends React.Component<NewRunParametersProps> {
         <div>{titleMessage}</div>
         {!!initialParams.length && (
           <div>
-            {initialParams.map((param, i) => (
-              <TextField
-                id={`newRunPipelineParam${i}`}
-                key={i}
-                variant='outlined'
-                label={param.name}
-                value={param.value || ''}
-                onChange={ev => handleParamChange(i, ev.target.value || '')}
-                style={{ maxWidth: 600 }}
-                className={commonCss.textField}
-              />
-            ))}
+            {initialParams.map((param, i) => {
+              return (
+                <EnhancedTextField
+                  key={i}
+                  param={param}
+                  index={i}
+                  handleParamChange={handleParamChange}
+                />
+              );
+            })}
           </div>
         )}
       </div>
     );
   }
 }
 
+interface EnhancedTextFieldProps {
+  param: ApiParameter;
```

Comment:
Because you are naming props as param and handleParamChange, shall we rename the component to ParamEditor?